subnet/subnet_test.go

```diff
@@ -402,3 +403,129 @@ func TestWatchNetworkRemoved(t *testing.T) {
 		t.Errorf("WatchNetwork produced wrong network: expected %s, got %s", expected, actual)
 	}
 }
+
+func TestAddReservation(t *testing.T) {
+	msr := newDummyRegistry()
+	sm := NewMockManager(msr)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	r := Reservation{
+		Subnet:   newIP4Net("10.4.3.0", 24),
+		PublicIP: ip.MustParseIP4("52.195.12.13"),
+	}
+	if err := sm.AddReservation(ctx, "_", &r); err == nil {
+		t.Fatalf("added reservation outside of configured network")
+	}
+
+	r.Subnet = newIP4Net("10.3.10.0", 24)
+	if err := sm.AddReservation(ctx, "_", &r); err != nil {
+		t.Fatalf("failed to add reservation: %v", err)
+	}
+
+	attrs := &LeaseAttrs{
+		PublicIP: r.PublicIP,
+	}
+	l, err := sm.AcquireLease(ctx, "_", attrs)
+	if err != nil {
+		t.Fatalf("failed to acquire subnet: %v", err)
+	}
+	if !l.Subnet.Equal(r.Subnet) {
+		t.Fatalf("acquired subnet is not the reserved one: expected %v, got %v", r.Subnet, l.Subnet)
+	}
+	if !l.Expiration.IsZero() {
+		t.Fatalf("acquired lease (prev reserved) has expiration set")
+	}
+
+	// Add the same reservation -- should succeed
```

Comment:
nit: I would suggest to move the last two subtests to line 426. So that we have reservation related tests all together. Then we try to acquireLease on a reservation.